hyperspy/axes.py

```diff
@@ -655,6 +720,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_index_changed(self):
+        self._update_attributes()
+        self.events.indices_changed.trigger(axes_manager=self)
+
+    def _on_view_changed(self):
+        self._update_attributes()
+        self.events.axes_changed.trigger()
```

Comment:
What I'm trying to do here is to create an event that triggers whenever a dependent signal would need to be recomputed (other than that the data has changed). I agree that the name of the event might be wrong though. An `axes_changed` event should trigger for any change to any attribute of any `DataAxis` (including the view, and name, and unit, etc.).

A better name for this event might be `transformation_changed`? Because the transform to map the coordinates of the signal space has changed?